src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -451,34 +441,33 @@ internal static bool CanHaveExecutableCodeBlock(ISymbol symbol)
 
         internal void ExecuteAndCatchIfThrows(DiagnosticAnalyzer analyzer, Action analyze)
         {
-            ExecuteAndCatchIfThrows(analyzer, analyze, _addExceptionDiagnostic, _continueOnAnalyzerException, _cancellationToken);
+            ExecuteAndCatchIfThrows(analyzer, analyze, _onAnalyzerException, _cancellationToken);
         }
 
         private static void ExecuteAndCatchIfThrows(
             DiagnosticAnalyzer analyzer,
             Action analyze,
-            Action<Diagnostic> addExceptionDiagnostic,
-            Func<Exception, DiagnosticAnalyzer, bool> continueOnAnalyzerException,
+            Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException,
             CancellationToken cancellationToken)
         {
             try
             {
                 analyze();
             }
-            catch (OperationCanceledException oce) when (continueOnAnalyzerException(oce, analyzer))
+            catch (OperationCanceledException oce)
             {
                 if (oce.CancellationToken != cancellationToken)
                 {
-                    // Add diagnostic for analyzer exception.
+                    // Diagnostic for analyzer exception.
                     var diagnostic = GetAnalyzerDiagnostic(analyzer, oce);
-                    addExceptionDiagnostic(diagnostic);
+                    onAnalyzerException(oce, analyzer, diagnostic);
                 }
             }
-            catch (Exception e) when (continueOnAnalyzerException(e, analyzer))
+            catch (Exception e)
             {
-                // Add diagnostic for analyzer exception.
+                // Diagnostic for analyzer exception.
                 var diagnostic = GetAnalyzerDiagnostic(analyzer, e);
-                addExceptionDiagnostic(diagnostic);
+                onAnalyzerException(e, analyzer, diagnostic);
```

Comment:
Is this a public extensibility point?   If so then wy should try catch that as well because it has the same issue of potentially throwing an exception through the compiler.